watcher/src/watcher.rs

```diff
@@ -219,6 +245,19 @@ impl Watcher {
             if !had_success {
                 return Ok(false);
             }
+
+            // If iteration limit was specified, check to see if it's been reached
+            if let Some(max_blocks_to_sync) = max_blocks_to_sync {
```

Comment:
Instead of an if statement, it might be easier to to use `max_blocks_to_sync.unwrap_or(u32::MAX)`. Btw, any reason why it is u32 and not usize?

Reply:
Sure, the unwrap_or is probably a better idea. Also it could be a usize, I'm just as yet a little fuzzy as to when usize vs. specifically sized unsigned integers.